CHANGELOG.md

```diff
@@ -4,6 +4,293 @@ All notable changes to Router will be documented in this file.
+
+By [@IvanGoncharov](https://github.com/IvanGoncharov) in https://github.com/apollographql/router/pull/6293
+
+### Overhead processing metrics should exclude subgraph response time when deduplication is enabled ([PR #6207](https://github.com/apollographql/router/pull/6207))
+
+The router's calculated overhead processing time has been fixed, where the time spent waiting for the subgraph response of a deduplicated request had been incorrectly included.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/6207
+
+### Fix demand control panic for custom scalars that represent non-GraphQL-compliant JSON ([PR #6288](https://github.com/apollographql/router/pull/6288))
+
+Previously, a panic could be triggered in the router's demand control plugin with the following schema:
+
+```
+scalar ArbitraryJson
+
+type MyInput {
+    json: ArbitraryJson
+}
+
+type Query {
+    fetch(args: MyInput): Int
+}
+```
+
+Then, submitting the query
+
+```
+query FetchData(: ArbitraryJson) {
+    fetch(args: {
+        json: 
+    })
+}
+```
+
+and variables
+
+```
+{
+    "myJsonValue": {
+        "field.with.dots": 1
+    }
+}
+```
+
+During scoring, the demand control plugin would attempt to convert the variable structure into a GraphQL-compliant structure requiring valid GraphQL names as keys. The dot characters in the keys however would cause a panic. 
+
+With this fix, only the GraphQL compliant part of the input object is scored, and the arbitrary JSON marked by the custom scalar is scored as an opaque scalar (similar to how built-ins like `Int` or `String` are processed).
+
+By [@tninesling](https://github.com/tninesling) in https://github.com/apollographql/router/pull/6288
+
+### Fix incorrect overriding of concrete type names with interface names when merging responses ([PR #6250](https://github.com/apollographql/router/pull/6250))
+
+When using `@interfaceObject`, differing pieces of data can come back with either concrete types or interface types depending on the source. Previously, receiving the data in a particular order could incorrectly result in the interface name of a type overwriting its concrete name.
+
+To make the response merging order-agnostic, the router now checks the schema to ensure concrete types are not overwritten with interfaces or less specific types.
+
+By [@tninesling](https://github.com/tninesling) in https://github.com/apollographql/router/pull/6250
+
+## 🛠 Maintenance
+
+### Query planner cache key improvements ([Issue #5160](https://github.com/apollographql/router/issues/5160))
+
+> [!IMPORTANT]
+> If you have enabled [Distributed query plan caching](https://www.apollographql.com/docs/router/configuration/distributed-caching/#distributed-query-plan-caching), this release changes the hashing algorithm used for the cache keys.  On account of this, you should anticipate additional cache regeneration cost when updating between these versions while the new hashing algorithm comes into service.
```

Comment:
the same is true for fragment generation. should this only be called out down here or at the top of the changelog, as it affects the whole release?

Reply:
ebf375a35